backend/database_test.go

```diff
@@ -0,0 +1,266 @@
+
+		//then
+		assert.Equal(t, "GETFROMDB:key " + key + " does not exist", err.Error())
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should return error when error occurred in getting value from database", func(t *testing.T) {
+
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(true)
+		database := Database{connection: client}
+		key := "key"
+		clientMock.ExpectGet(key)
+
+		//when
+		_, err := database.GetFromDB(key)
+
+		//then
+		assert.Contains(t, err.Error(), "GETFROMDB:error: ")
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should return error when key exists but value is empty", func(t *testing.T) {
+
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(true)
+		database := Database{connection: client}
+		key := "key"
+		clientMock.ExpectGet(key).SetVal("")
+
+		//when
+		val, err := database.GetFromDB(key)
+
+		//then
+		assert.Equal(t, "", val)
+		assert.Equal(t, "GETFROMDB:for key " + key + " value is empty", err.Error())
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should acquire data correctly using GetFromDB function", func(t *testing.T) {
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(true)
+		database := Database{connection: client}
+		key, value := "key", "value"
+		clientMock.ExpectGet(key).SetVal(value)
+
+		//when
+		val, err := database.GetFromDB(key)
+		assert.NoError(t, err)
+
+		//then
+		assert.Equal(t, value, val)
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+}
+
+func TestGetAllKeysDB(t *testing.T) {
+	t.Run("should return error when connection is not initialized", func(t *testing.T) {
+		//given
+		_, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(false)
+		database := Database{connection: nil}
+
+		//when
+		_, err := database.GetAllKeys()
+
+		//then
+		assert.Error(t, err, errors.New("GETALLKEYS: connection not initialized"))
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should return error when error occurred in getting keys from database", func(t *testing.T) {
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(false)
+		database := Database{connection: client}
+		clientMock.ExpectKeys("*")
+
+		//when
+		val, err := database.GetAllKeys()
+
+		//then
+		assert.Error(t, err)
+		var n []string = nil
+		assert.Equal(t, n, val)
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should acquire all keys using the GetAllKeys function", func(t *testing.T) {
```

Comment:
Do not state verbosely `using the GetAllKeys function` because the function name can change - it is only implementation, test descriptions should be high-level and generic